File: contrib/integration/session/java/com/cppcms/session/SessionPool.java
package com.cppcms.session;

import com.sun.jna.Pointer;


public class SessionPool extends SessionBase {

	 private SessionPool()
	 {
		API.init();
		d=API.api.cppcms_capi_session_pool_new();
	 }
	 public static SessionPool openFromConfig(String file) 
	 {
		 SessionPool p=new SessionPool();
		 API.api.cppcms_capi_session_pool_init(p.d,file);
		 p.check();
		 return p;
	 }
	 public static SessionPool openFromJson(String json)
	 {
		 SessionPool p=new SessionPool();
		 API.api.cppcms_capi_session_pool_init_from_json(p.d,json);
		 p.check();
		 return p;
	 }
	 public Session getSession()
	 {
		 return new Session(d);
	 }
	 void doClose()
	 {
		API.api.cppcms_capi_session_pool_delete(d);
	 }
};

File: contrib/integration/session/asp.net/SessionPool.cs
using System;
namespace CppCMS.Session  {
public class SessionPool : SessionBase {
	private SessionPool()
	{
		d=API.session_pool_new();
	}
	override public void Close()
	{
		if(d!=IntPtr.Zero) {
			API.session_pool_delete(d);
			d=IntPtr.Zero;
		}
	}
	public static SessionPool FromConfig(string path)
	{
		SessionPool p = new SessionPool();
		API.session_pool_init(p.d,tb(path));
		p.check();
		return p;
	}
	public static SessionPool FromJSON(string json)
	{
		SessionPool p = new SessionPool();
		API.session_pool_init_from_json(p.d,tb(json));
		p.check();
		return p;
	}
	public Session Session()
	{
		return new Session(this);
	}
}
} // namespace

